lib/spack/spack/test/cmd/concretize.py

```diff
@@ -58,3 +60,23 @@ def test_concretize_root_test_dependencies_are_concretized(unify, mutable_mock_e
 
         data = e._to_lockfile_dict()
         assert data["spack"]["version"] == spack_version
+
+
+@pytest.mark.regression("48254")
+@pytest.mark.parametrize("unify", unification_strategies)
+def test_concretize_fresh_respected_with_env(unify, mutable_mock_env_path, monkeypatch):
+    """Check that root test dependencies are concretized."""
+    env("create", "test")
+
+    def mock_reusable_fresh(self, specs):
+        assert self.reuse_strategy == asp.ReuseStrategy.NONE
+        return []
+
+    monkeypatch.setattr(asp.ReusableSpecsSelector, "reusable_specs", mock_reusable_fresh)
```

Comment:
This is the type of test isn't helpful if we ever refactor. I would test the behavior not the implementation.

Reply:
I see your point. FWIW I already tested back porting this across api breaks. It's not a hard test to update but it can require changing the monnkepatch method as you're alluding to. This test reproduces the regression which is why I designed it that way and marked it as a regression test. The regression being run --fresh from an active env that also sets the reuse through config.

Do we have a policy/ preference on regression over true unit-test? Since we lack true API's in the internals of spack it's messy. The only true api contract seems to be CLI which is why I wrote it this way.

You should write a unit test guide for the project since my tests rarely live up to your standards 😜.